src/aggregate/aggregate_exec.c

```diff
@@ -619,6 +619,8 @@ static void sendChunk_Resp3(AREQ *req, RedisModule_Reply *reply, size_t limit,
     } else if (rc == RS_RESULT_ERROR) {
       // Non-fatal error
       RedisModule_Reply_SimpleString(reply, QueryError_GetError(req->qiter.err));
+    } else if (req->qiter.err) {
+      RedisModule_Reply_SimpleString(reply, "Max prefix expansions reached");
```

Comment:
Maybe this should be a separate element in the array?
Can't timeout happen and also max prefix expansions reached happen at the same time?

Reply:
I agree in general that an array makes sense here, as they indeed can both occur for the same execution.
The thing is, that this change would break the response, such that we won't be able to release this in a patch.

Moreover, they have the same effect on the results - truncating them, and you don't know which results didn't come from which error as well. I think that timeout has higher precedence thus it should be reported first, and we should extend this to be an array once we release a breaking change to our response.

Reply:
Looking at the code I think it is already an array, no?
```    RedisModule_Reply_ArrayEnd(reply); // >warnings```

Reply:
It is, but populating it with more warnings may be more breaking in a way. 
We plan to have a continuation PR to this PR which will refactor the `QueryError` to be `QueryStatus` and contain a similar API for the warnings as there currently is for the errors. We will also make the warnings array actually able to reply with multiple warnings if breakage will allow.
If this won't be an urgent PR we may also do this here, but seems that we want to have a quick fix that will later be improved.